crates/pathfinder/src/state/ethereum.rs

```diff
@@ -0,0 +1,137 @@
+use std::time::Duration;
+
+use anyhow::Context;
+use tokio::sync::{mpsc, oneshot};
+use web3::{
+    types::{BlockId, BlockNumber},
+    Transport, Web3,
+};
+
+use crate::{
+    core::StarknetBlockNumber,
+    ethereum::{
+        log::{FetchError, StateUpdateLog},
+        state_update::state_root::StateRootFetcher,
+        Chain,
+    },
+};
+
+/// The set of events which may be emitted by [update_ethereum_state].
+enum Event {
+    /// Request the [StateUpdateLog] for the [StarknetBlockNumber]. The result should be returned
+    /// using the [oneshot::Sender] channel.
+    GetUpdate(StarknetBlockNumber, oneshot::Sender<Option<StateUpdateLog>>),
+    /// Batch of new [StateUpdateLog].
+    Updates(Vec<StateUpdateLog>),
+    /// An L1 reorg was detected. Contains the reorg tail.
+    /// i.e. all blocks __up-to-and-including__ this value are now invalid.
+    Reorg(StarknetBlockNumber),
+}
+
+/// Syncs L1 state update logs. Emits [Events](Event) which should be handled
+/// to update storage and respond to queries.
+async fn sync_ethereum_state<Tr: Transport>(
+    transport: Web3<Tr>,
```

Comment:
Should this also be a separate channel? So that the web3 thingies should also be part of a separate operation?

I guess I'm unclear on which things should be externalised or not.

Reply:
Made this a separate channel.